packages/bpk-component-button/stories.js

```diff
@@ -46,80 +53,97 @@ const cssModules = (styles = {}) => className =>
 
-const ButtonStory = ({ className, ...rest }: { className: ?string }) => (
-  <div
-    className={[getClassName('bpk-button-story-wrapper'), className].join(' ')}
-  >
-    &nbsp;
-    <BpkButton onClick={action('button clicked')} {...rest}>
-      Button
-    </BpkButton>
-    &nbsp;
-    <BpkButton disabled onClick={action('THIS SHOULD NOT HAPPEN')} {...rest}>
-      Disabled
-    </BpkButton>
-    &nbsp;
-    <BpkButton large onClick={action('large button clicked')} {...rest}>
-      Button
-    </BpkButton>
-    &nbsp;
-    <BpkButton
-      large
-      disabled
-      onClick={action('THIS SHOULD NOT HAPPEN')}
-      {...rest}
-    >
-      Disabled
-    </BpkButton>
-    &nbsp;
-    <BpkButton iconOnly onClick={action('iconOnly button clicked')} {...rest}>
-      <AlignedSmallLongArrowRightIcon />
-    </BpkButton>
-    &nbsp;
-    <BpkButton
-      iconOnly
-      large
-      onClick={action('large iconOnly button clicked')}
-      {...rest}
-    >
-      <AlignedLargeLongArrowRightIcon />
-    </BpkButton>
-    &nbsp;
-  </div>
-);
+const wrapButton = Component => {
+  function wrapComponent({ className, ...rest } = { className: '' }) {
+    return (
+      <div
+        className={[getClassName('bpk-button-story-wrapper'), className].join(
+          ' ',
+        )}
+      >
+        &nbsp;
+        <Component onClick={action('button clicked')} {...rest}>
+          Button
+        </Component>
+        &nbsp;
+        <Component
+          disabled
+          onClick={action('THIS SHOULD NOT HAPPEN')}
+          {...rest}
+        >
+          Disabled
+        </Component>
+        &nbsp;
+        <Component large onClick={action('large button clicked')} {...rest}>
+          Button
+        </Component>
+        &nbsp;
+        <Component
+          large
+          disabled
+          onClick={action('THIS SHOULD NOT HAPPEN')}
+          {...rest}
+        >
+          Disabled
+        </Component>
+        &nbsp;
+        <Component
+          iconOnly
+          onClick={action('iconOnly button clicked')}
+          {...rest}
+        >
+          <AlignedSmallLongArrowRightIcon />
+        </Component>
+        &nbsp;
+        <Component
+          iconOnly
+          large
+          onClick={action('large iconOnly button clicked')}
+          {...rest}
+        >
+          <AlignedLargeLongArrowRightIcon />
+        </Component>
+        &nbsp;
+      </div>
+    );
+  }
 
-ButtonStory.defaultProps = { className: null };
+  return wrapComponent;
+};
 
 storiesOf('bpk-component-button', module)
-  .add('Primary', () => <ButtonStory primary />)
-  .add('Secondary', () => <ButtonStory secondary />)
-  .add('Destructive', () => <ButtonStory destructive />)
-  .add('Link button', () => <ButtonStory link />)
-  .add('Featured', () => <ButtonStory featured />)
-  .add('Outline', () => (
-    <ButtonStory outline className={getClassName('bpk-outline-layout')} />
-  ))
+  .add('Primary', () => wrapButton(BpkButtonPrimary)())
```

Comment:
Any reason for the HOC here? I would keep the old style. Something like `<ButtonStory wrapper={BpkButtonPrimary} />`

Reply:
I suggested a HOC, but yeah your idea works too

Reply:
I have reverted back to the old way and modified it.